core/lib/eth_client/src/lib.rs

```diff
@@ -168,13 +185,18 @@ pub trait EthInterface: Sync + Send {
 /// 2. Consider adding the "unbound" version to the `EthInterface` trait and create a default method
 ///   implementation that invokes `contract` / `contract_addr` / `sender_account` methods.
 #[async_trait]
-pub trait BoundEthInterface: AsRef<DynClient<L1>> + 'static + Sync + Send + fmt::Debug {
+pub trait BoundEthInterface<Net: Network>:
```

Comment:
- Just intuitively, `Net` should be an associated type rather than a type param, like in `DynClient`. I.e., an implementation has the "network" property / cannot implement `BoundEthInterface` for multiple networks.
- Dumb question: Is this trait ever used for `Net != L1`? AFAICT, the only instance is the `MockSettlementLayer` impl, but the workspace still compiles after being narrowed to `impl BoundEthInterface<L1> for MockSettlementLayer<L1>`. So it looks like this parameter is redundant. Would `BoundEthInterface` be used with different networks in the future?

Reply:
It will be also used for Net = L2 with the next PRs

Reply:
> but the workspace still compiles after being narrowed to impl BoundEthInterface<L1> for MockSettlementLayer<L1>

The tests don't. And yes as @tomg10 said we will use it for L2

Reply:
@slowli the problem right now is we want both `DynClient<L1>`-based and `DynClient<L2>`-based structs to be able to implement this trait. I understand the logic that a struct can not implement at the same time both `BoundEthInterface<L1>` and `BoundEthInterface<L2>`, but in this case `DynClient<L1>` will be a requirement for the struct that wanted to implement `BoundEthInterface<L2>` which is hard to achieve. 

So it is easier to say please provide some `DynClient<Net>` and so we have to make `Net` a template param